test/Semantics/test_modfile.sh

```diff
@@ -26,7 +26,7 @@ for src in "$@"; do
   ) > $actual_files
   expected_files=$(sed -n 's/^!Expect: \(.*\)/\1/p' $src | sort)
   extra_files=$(echo "$expected_files" | comm -23 $actual_files -)
-  if [[ ! -z "$extra_files" ]]; then
+  if [[ ! -z "$extra_files" && ! -z "$expected_files" ]]; then
```

Comment:
Why is this change needed?

Reply:
It's not, I thought I was getting some bizarre errors regarding files coming up in the extra_files var, but it was an outdated compiler.

Reply:
Okay, turns out I was mistaking this change for another. I removed this and getsymbols02.f90 passed, but when I cleared my cache it failed.

getsymbols02-a.f90 complains `Unexpected .mod files produced: mm2a.mod`, but that seems like valid behaviour to me. Therefore, since all generic tests that use modfiles, never have the produced modfile in their expected files (when it should), I added this to make sure those tests still pass.

Reply:
That test is supposed to be run with test_any.sh, not test_modfile.sh.